sky/backends/backend_utils.py

```diff
@@ -1826,6 +1835,20 @@ def run_ray_status_to_check_ray_cluster_healthy() -> bool:
         return record
 
     # All cases below are transitioning the cluster to non-UP states.
+
+    if len(node_statuses) == 0:
+        time_since_launch = time.time() - record['launched_at']
```

Comment:
`launched_at` might be updated by non launching command. We need to double check if it makes sense to always use `launched_at`

Reply:
Looked into this:
1. We will set `launched_at` every time we run `sky launch` (maybe not if you use `--fast`, but anyway). So this is not ideal since we only really care about the initial cluster creation.
2. We will set it twice - once when initially setting the cluster to INIT, and once once provisioning is finished and we set the cluster to UP.

2) is kind of ok, since we won't run the check if the previous cluster state is UP. 1) is a bit more problematic, looking into what I can do here.

Reply:
I think to fix 1) we can use usage_intervals from the cluster history table instead. Working on a patch for this.

Reply:
Updated this

Reply:
Going to revert to using launched_at as it's a more stable code path. It may result in checking this a bit more often but not much.